crawlers/mooncrawl/mooncrawl/moonworm_crawler/cli.py

```diff
@@ -198,7 +261,24 @@ def handle_historical_crawl(args: argparse.Namespace) -> None:
         )
         logger.info(f"Last labeled block: {last_labeled_block}")
 
-        start_block = args.start
+        addresses_deployment_blocks = None
+
+        # get set of addresses from event jobs and function call jobs
+        if args.find_deployed_blocks:
+            addresses_set = set()
+            for job in filtered_event_jobs:
+                addresses_set.update(job.contracts)
+            for function_job in filtered_function_call_jobs:
+                addresses_set.add(function_job.contract_address)
+
+            if args.start is None:
+                start_block = web3.eth.blockNumber - 1
+
+            addresses_deployment_blocks = find_all_deployed_blocks(
+                web3, list(addresses_set)
+            )
+            end_block = min(addresses_deployment_blocks.values())
```

Comment:
According to func `find_all_deployed_blocks` it could return empty dict `{}` so min resuction will throw exception. Maybe better to not return empty dict but throw exception in func `find_all_deployed_blocks`?

Reply:
Add that way.
```python
          if len(addresses_deployment_blocks) == 0:
              raise ValueError(
                  "No addresses found in the blockchain. Please check your addresses and try again"
              )
```